src/classes/UnitOfWork.cls

```diff
@@ -0,0 +1,78 @@
+/*
+    Copyright (c) 2021 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @description Unit of Work in NPSP.
+*/
+public inherited sharing class UnitOfWork {
```

Comment:
Yes, Victor, good call!

As demonstrated by this work, NPSP needs a [Unit of Work](https://trailhead.salesforce.com/en/content/learn/modules/apex_patterns_sl/apex_patterns_sl_learn_uow_principles) so that Domains can be unit-tested.   I'd prefer to bring in the [Unit of Work from fflib](https://github.com/wimvelzeboer/fflib-apex-common/blob/master/sfdx-source/apex-common/main/classes/fflib_SObjectUnitOfWork.cls) along with its [full test suite](https://github.com/wimvelzeboer/fflib-apex-common/blob/master/sfdx-source/apex-common/test/classes/fflib_SObjectUnitOfWorkTest.cls) rather than building this one out further.  Curious to hear the teams thoughts.

Reply:
I'd like to see the fflib version of UoW get pulled in too, already built and has test coverage even for the stuff we wouldn't be using quite yet. 👍

Reply:
(Blocking!!!!) I think the expectation is clear that we need test coverage for anything we are shipping to customers, even if temporary. If we aren't ready to bring in the fflib UoW infrastructure before me merge this to the feature branch, I think we need a test class here (even if relatively basic).

Reply:
This class/code is definitely covered, it just doesn't have its own explicit test class.  My opinion is that we shouldn't spend time writing new dedicated tests for this utility.  It is covered by implicit tests.  And if/when we are able to use fflib UoW, then NPSP will have a full, robust UoW with its own dedicated tests.
![2021-05-18_05-23-37](https://user-images.githubusercontent.com/46458081/118658816-99082e80-b7a1-11eb-8cf6-317c4028e4b1.png)

Reply:
Sold! Thanks for laying that out for me